pkgdb/src/buildenv/realise.cc

```diff
@@ -250,12 +251,36 @@ createFloxEnv( nix::EvalState &     state,
 
       if ( ! package_drv.has_value() )
         {
-          throw FloxException( "Failed to get derivation for package '"
-                               + nlohmann::json( package ).dump() + "'" );
+          throw PackageEvalFailure( "Failed to get derivation for package '"
+                                    + nlohmann::json( package ).dump() + "'" );
         }
 
-      auto packagePath
-        = state.store->printStorePath( package_drv->queryOutPath() );
+      std::string packagePath;
+      try
+        {
+          packagePath
+            = state.store->printStorePath( package_drv->queryOutPath() );
+        }
+      catch ( const nix::Error & e )
+        {
+
+          if ( e.info().msg.str().find(
+                 "is not available on the requested hostPlatform:" )
+               != std::string::npos )
+            {
+              throw PackageUnsupportedSystem(
+                "'" + pId
+                  + "' is not available on the requested hostPlatform ('"
```

Comment:
Suggestion: since this could end up user facing it should probably say something more familiar like "not available for this system".